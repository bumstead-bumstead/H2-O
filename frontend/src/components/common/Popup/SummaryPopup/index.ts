export { default as SummaryPopup } from './SummaryPopup';
