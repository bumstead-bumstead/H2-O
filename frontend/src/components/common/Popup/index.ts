export { default as Popup } from './Popup';
export { default as SummaryPopup } from './SummaryPopup/SummaryPopup';
export { default as ConfirmPopup } from './ConfirmPopup/ConfirmPopup';
