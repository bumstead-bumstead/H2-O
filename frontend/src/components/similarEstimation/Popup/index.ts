export { default as ConfirmClosePopup } from './ConfirmClosePopup';
export { default as ConfirmOptionPopup } from './ConfirmOptionPopup';
