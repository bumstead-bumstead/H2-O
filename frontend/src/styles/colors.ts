export const COLORS = {
  // primary
  primary100: '#CCD7E5',
  primary200: '#99B0CA',
  primary300: '#6688B0',
  primary400: '#336196',
  primary500: '#00397B',
  primary600: '#002E63',
  primary700: '#00224A',
  primary800: '#001731',
  primary900: '#000B19',

  // white
  white: '#FFFFFF',

  // grayscale
  gray50: '#F7F8F9',
  gray100: '#EDEDEE',
  gray200: '#DADCDD',
  gray300: '#C8CACC',
  gray400: '#B6B8BB',
  gray500: '#A3A6AA',
  gray600: '#838588',
  gray700: '#626466',
  gray800: '#414344',
  gray900: '#212122',

  // other colors
  activeBlue: '#00AAD2',
  activeBlue2: '#00C3F0',
  skyBlue: '#A2C7E7',
  skyBlue2: '#e3eef8',
  hoverBg: '#A2C7E70D',
  skyBlueCardBg: '#A2C7E74D',
  cardBg: '#00AAD21A',
  blueBg: '#F4F5F7',
  sand: '#B67B5E',
  sand2: '#E4DCD3',
  lightSand: '#F6F3F2',
  priceSummaryHead: '#343434',
  hashTagBg: '#75757580',
  footerBg: '#FFFFFFE5',

  // dimmed background
  dimmed: '#1F1F1FB3',
} as const;

export type ColorType = typeof COLORS;
export type Colors = keyof ColorType;
