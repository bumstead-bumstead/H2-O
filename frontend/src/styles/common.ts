export const FLEX = {
  flexDefault: {
    display: 'flex',
  },
  flexCenterCol: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  flexCenterRow: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  flexBetweenCol: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
  flexBetweenRow: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  flexEndRow: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
  flexEndCol: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-end',
  },
} as const;

export type FlexType = typeof FLEX;
export type FlexVariant = keyof FlexType;
