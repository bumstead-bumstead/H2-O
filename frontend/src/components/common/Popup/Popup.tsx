import { HTMLAttributes, MouseEventHandler, PropsWithChildren } from 'react';
import styled from '@emotion/styled';
import { Dimmed, Portal } from '@/components/common';

interface Props extends HTMLAttributes<HTMLDivElement> {
  size?: 'small' | 'large';
  handleClickDimmed: MouseEventHandler<HTMLDivElement>;
}

function Popup({ children, size = 'small', handleClickDimmed, ...restProps }: PropsWithChildren<Props>) {
  return (
    <Portal>
      <Dimmed data-testid='dimmed' onClick={handleClickDimmed} {...restProps} />
      <PopupContainer role='dialog' size={size} {...restProps}>
        {children}
      </PopupContainer>
    </Portal>
  );
}

export default Popup;

export const PopupContainer = styled.div<Pick<Props, 'size'>>`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background: ${({ theme }) => theme.colors.white};
  border-radius: ${({ size }) => (size === 'small' ? 4 : 8)}px;
  width: ${({ size }) => (size === 'small' ? 336 : 850)}px;
  height: ${({ size }) => (size === 'small' ? 200 : 520)}px;
  overflow: hidden;
  z-index: 20;
`;
