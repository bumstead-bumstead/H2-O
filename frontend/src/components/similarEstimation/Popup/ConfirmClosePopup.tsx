import { css } from '@emotion/react';
import { ConfirmPopup, Flex, Typography } from '@/components/common';
import { useSafeContext } from '@/hooks';
import { SlideContext } from '@/providers/SlideProvider';

interface Props {
  closeSimilarEstimationPopup: () => void;
  closePopup: () => void;
  isEmptyOption: boolean;
  optionCnt?: number;
}

function ConfirmClosePopup({ closePopup, closeSimilarEstimationPopup, isEmptyOption, optionCnt }: Props) {
  const { setCurrentSlide } = useSafeContext(SlideContext);

  const handleClickConfirmButton = () => {
    closePopup();
    closeSimilarEstimationPopup();
    setCurrentSlide(5);
  };

  return (
    <ConfirmPopup
      hasCancelButton={true}
      confirmButtonLabel={isEmptyOption ? '종료' : '확인'}
      handleClickCancelButton={closePopup}
      handleClickConfirmButton={handleClickConfirmButton}
      css={PopupPosition}
    >
      {isEmptyOption ? (
        '유사견적 페이지를 닫으시겠습니까?'
      ) : (
        <Flex flexDirection='column' gap={16}>
          <Typography color='primary500'>선택된 옵션 {optionCnt}개</Typography>
          <p>
            선택된 옵션이 있습니다. <br />
            지금 서비스 종료 시 저장되지 않습니다.
          </p>
        </Flex>
      )}
    </ConfirmPopup>
  );
}

export default ConfirmClosePopup;

const PopupPosition = css`
  z-index: 25;
`;
