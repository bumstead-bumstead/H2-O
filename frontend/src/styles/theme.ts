import { COLORS } from './colors';
import { FLEX } from './common';
import { TYPOGRAPHY } from './typography';

export const theme = {
  colors: COLORS,
  flex: FLEX,
  typography: TYPOGRAPHY,
};
