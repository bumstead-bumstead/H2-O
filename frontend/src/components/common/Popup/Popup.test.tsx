import Popup from './Popup';
import { fireEvent, render, screen } from '@/tests/test-util';
import { COLORS } from '@/styles/colors';

describe('Popup 컴포넌트 테스트', () => {
  beforeEach(() => {
    const portalRoot = global.document.createElement('div');
    portalRoot.setAttribute('id', 'portal-root');

    const body = global.document.querySelector('body');
    if (!body) throw new Error('portal-root의 body를 찾을 수 없습니다.');
    body.appendChild(portalRoot);
  });

  const handleClickedDimmend = vi.fn();

  it('Popup 컴포넌트가 정상적으로 렌더링 된다.', () => {
    render(<Popup size='small' handleClickDimmed={handleClickedDimmend} />);

    const portalRoot = document.getElementById('portal-root');
    expect(portalRoot).toBeInTheDocument();
  });

  it('Popup 컴포넌트를 렌더링하면 기본적으로 dimmmed와 작은 흰색 팝업창이 생성된다.', () => {
    render(<Popup size='small' handleClickDimmed={handleClickedDimmend} />);

    expect(screen.getByTestId('dimmed')).toBeInTheDocument();
    expect(screen.getByRole('dialog')).toBeInTheDocument();
    expect(screen.getByRole('dialog')).toHaveStyle(`background:${COLORS.white}`);
    expect(screen.getByRole('dialog')).toHaveStyle('border-radius:4px; width: 336px; height:200px;');
  });

  it('Popup size 속성을 통해서 흰색 팝업창의 크기를 조절할 수 있다.', () => {
    render(<Popup size='large' handleClickDimmed={handleClickedDimmend} />);

    expect(screen.getByRole('dialog')).toBeInTheDocument();
    expect(screen.getByRole('dialog')).toHaveStyle(`background:${COLORS.white}`);
    expect(screen.getByRole('dialog')).toHaveStyle('border-radius:8px; width: 850px; height:520px;');
  });

  it('Popup의 Dimmed 영역을 클릭하면 클릭 이벤트가 실행된다.', () => {
    render(<Popup size='large' handleClickDimmed={handleClickedDimmend} />);

    const dimmedElement = screen.getByTestId('dimmed');
    fireEvent.click(dimmedElement);

    expect(handleClickedDimmend).toBeCalledTimes(1);
  });

  describe('Popup 컴포넌트 스냅샷 테스트', () => {
    it('Popup 컴포넌트 size가 small인 컴포넌트 스냅샷', () => {
      render(<Popup size='small' handleClickDimmed={handleClickedDimmend} />);

      expect(document.getElementById('portal-root')).toMatchSnapshot();
    });

    it('Popup 컴포넌트 size가 large인 컴포넌트 스냅샷', () => {
      render(<Popup size='large' handleClickDimmed={handleClickedDimmend} />);

      expect(document.getElementById('portal-root')).toMatchSnapshot();
    });
  });
});
