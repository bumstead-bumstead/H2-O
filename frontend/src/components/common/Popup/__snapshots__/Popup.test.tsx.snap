// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popup 컴포넌트 테스트 > Popup 컴포넌트 스냅샷 테스트 > Popup 컴포넌트 size가 large인 컴포넌트 스냅샷 1`] = `
.emotion-0 {
  background-color: #1F1F1FB3;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  -webkit-backdrop-filter: blur(6px);
  backdrop-filter: blur(6px);
  z-index: 10;
}

.emotion-1 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  background: #FFFFFF;
  border-radius: 8px;
  width: 850px;
  height: 520px;
  overflow: hidden;
  z-index: 20;
}

<div
  id="portal-root"
>
  <div
    class="emotion-0"
    data-testid="dimmed"
  />
  <div
    class="emotion-1"
    role="dialog"
  />
</div>
`;

exports[`Popup 컴포넌트 테스트 > Popup 컴포넌트 스냅샷 테스트 > Popup 컴포넌트 size가 small인 컴포넌트 스냅샷 1`] = `
.emotion-0 {
  background-color: #1F1F1FB3;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  -webkit-backdrop-filter: blur(6px);
  backdrop-filter: blur(6px);
  z-index: 10;
}

.emotion-1 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  background: #FFFFFF;
  border-radius: 4px;
  width: 336px;
  height: 200px;
  overflow: hidden;
  z-index: 20;
}

<div
  id="portal-root"
>
  <div
    class="emotion-0"
    data-testid="dimmed"
  />
  <div
    class="emotion-1"
    role="dialog"
  />
</div>
`;
